### PR TITLE
fix: make theme logic safe and add better e2e tests

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

**Tldr:**
1. The bug is NOT in production
2. Setting an app theme to ”system” in my-account/appearance sets the value of app-theme in localStorage to "on"—it should be set to ”system”
3. Don’t even know where “on” is coming from. I’m debugging.

My recent [PR](https://github.com/calcom/cal.com/pull/16793) that’s completely replaced pages router with app router for settings/my-account pages introduced this bug:

https://www.loom.com/share/7b43f7a3a47e4d83b1be3b14fde89248?sid=519c6e6a-0cf1-44d8-8e4d-1dbb82d1d3b9

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

**Comment by hbjORbj** [comment]
This needs to be included in the Tuesday release @zomars

**Review by sean-brydon — APPROVED** [review]
LGTM! This fixes the issue on myside. Nice job on the tests
